R/config-log.R

```diff
@@ -0,0 +1,129 @@
+###############################
+# Fields from bbi_config.json
+###############################
+
+#' Parse babylon configs to log
+#'
+#' Parses `bbi_config.json` files into a log tibble.
+#' This file is created by babylon, in the model output folder.
+#' It stores metadata about the execution of a model run,
+#' including md5 hashes of the model file (control stream) and data file at the time the model was run.
+#'
+#' @details
+#' `config_log()` will return a tibble with one row per `bbi_config.json` found.
+#'
+#' `add_config()` takes a `bbi_run_log_df` tibble (the output of `run_log()`) as its input,
+#' searches for a `bbi_config.json` for each row in the input tibble,
+#' and joins on the data from the relevant config, if found.
+#' The returned tibble will have all of its input columns, plus all of the columns returned from `config_log()`
+#'
+#' @seealso `run_log()`
+#' @param .base_dir Base directory to look in for models. Defaults to `get_model_directory()`, and falls back to `getwd()` if `get_model_directory()` returns `NULL`.
```

Comment:
I think I follow you here. In reality, it will use `getwd()` when _either_ `NULL` is passed directly to `.base_dir` _or_ when `rbabylon.model_directory` is `NULL` (because `get_model_directory()` will return `NULL` in that case).

```
get_model_directory <- function() {
  .mod_dir <- getOption("rbabylon.model_directory")
  if (is.null(.mod_dir)) {
    return(NULL)
  }
...
```

Anyway, I would support adding that message if you think we should, but I'm not sure it's necessary. Especially given that it tells you that's gonna happen in the docs. I can clarify the docs though, if that helps.

Reply:
IMO fallback behavior should be very carefully considered, especially given we can provide an essential "fallback" of a default parameter behavior. If we do provide a fallback, it should be opinionated and intuitive. In this case, the working directory as the modeling directory is generally "never" the situation when running scripts as people keep scripts/project in separate directories than models, so config log instead would likely lead them astray in such a situation. Rather than being caught and a helpful error message along the lines of "no directory set", it would likely cause misbehavior (eg a failure to find relevant models in the wd), or worse, could mis-detect some models there and give spurious results. 

I would say ~ option 1 that the assertion of an actual user specified directory that is a string associated with a real path (eg don't want directory "")

Reply:
Yea when I think more about it, this makes sense. User should have to either pass an explicit directory or have rbabylon.model_directory set. I'm good with that.